infra/embedding_step_functions/unified_embedding/handlers/compaction.py

```
@@ -21,8 +21,16 @@
 
 # Import receipt_dynamo for proper DynamoDB operations
 from receipt_dynamo.data.dynamo_client import DynamoClient
+from receipt_dynamo.constants import ChromaDBCollection
 from receipt_label.utils.lock_manager import LockManager
 
+try:
+    from receipt_label.utils.chroma_s3_helpers import upload_snapshot_with_hash
+    HASH_UPLOAD_AVAILABLE = True
+except ImportError:
+    HASH_UPLOAD_AVAILABLE = False
+    logger.warning("Hash-enabled upload not available, using legacy upload")
```

Ruff (0.12.2)

Undefined name `logger`

(F821)